packages/grafana-data/src/themes/colorManipulator.ts

```diff
@@ -32,19 +32,23 @@ export function hexToRgb(color: string) {
   color = color.slice(1);
 
   const re = new RegExp(`.{1,${color.length >= 6 ? 2 : 1}}`, 'g');
-  let colors = color.match(re);
+  const result = color.match(re);
+  if (!result) {
+    return '';
+  }
+  let colors = Array.from(result);
```

Comment:
`RegExpMatchArray` is now typed to always have a `0` index, which caused problems with `arr.map` because `map` returns just an `Array` (with no guarenteed `0` index), and so cannot be assigned back to `RegExpMatchArray`

I slightly refactored this function to:

 - bail early if the regex didn't find any matches
 - convert the regex result to an explict array using `Array.from`
 - move `segments` out of the template string to simplify the code a bit

Reply:
No, it's a part of the public API, which we cannot ship breaking changes to in patch/point releases at least.

Reply:
ah yeah. kinda strange that themes are in grafana-data and not grafana-ui 🤷

Reply:
How The Gods decide what goes into grafana-data/ui/runtime will always be a mystery :)